vars/named_vars.go

```diff
@@ -12,7 +12,7 @@ func (m NamedVariables) Get(ref Reference) (interface{}, bool, error) {
 	}
 
 	if vars, ok := m[ref.Source]; ok {
-		return vars.Get(ref)
+		return vars.Get(ref.WithoutSource())
```

Comment:
Just wondering if it is necessary to remove the source from this `vars.Get` call?

Reply:
Since `NamedVariables` keeps track of multiple other implementations, `Source` is actually only meaningful to `NamedVariables` and nothing else. 

Let's say during a build we end up with a build var that looks like this:
```
NamedVar {
  "foo": StaticVar{
    "secret": "value"
  }
  "bar": Tracker{
    Vault: ....
  }
}
```

If we try to get `((foo:secret))`, it should pass `((secret))` to the StaticVar (as part of this PR, StaticVar will actually return false if the source is non-empty). Likewise if we try `((bar:path.field))`, it'll only pass `((path.field))` to Tracker/Vault.

Reply:
Just to make sure I completely understand, this change is purely refactoring and has no real user impact changes/behavioural changes right? Because the `NamedVariables` is only used for var source implementations so it will never contain a `StaticVar` which is used for build variables.